api/app/invitationletter/api.py

```diff
@@ -0,0 +1,124 @@
+
+        invitation_template = None
+
+        if offer.accommodation_award and offer.accepted_accommodation_award \
+                and offer.travel_award and offer.accepted_travel_award:
+            invitation_template = db.session.query(InvitationTemplate).filter(
+                InvitationTemplate.event_id == offer.event_id).filter(
+                InvitationTemplate.send_for_both_travel_accommodation).first()
+
+        elif offer.travel_award and offer.accepted_travel_award:
+            invitation_template = db.session.query(InvitationTemplate).filter(
+                InvitationTemplate.event_id == offer.event_id).filter(
+                InvitationTemplate.send_for_travel_award_only).first()
+
+        elif offer.accommodation_award and offer.accepted_accommodation_award:
+            invitation_template = db.session.query(InvitationTemplate).filter(
+                InvitationTemplate.event_id == offer.event_id).filter(
+                InvitationTemplate.send_for_accommodation_award_only).first()
+
+        elif (not offer.accommodation_award) and (not offer.travel_award):
+            invitation_template = db.session.query(InvitationTemplate)\
+                .filter(not InvitationTemplate.send_for_both_travel_accommodation)\
+                .filter(not InvitationTemplate.send_for_travel_award_only)\
+                .filter(not InvitationTemplate.send_for_accommodation_award_only).first()
+
+        if invitation_template:
+            template_url = invitation_template.template_path
+
+            user = db.session.query(AppUser).filter(
+                AppUser.id == user_id).first()
+
+            if not user:
+                return errors.USER_NOT_FOUND
+
+            country_of_residence = db.session.query(Country).filter(Country.id == user.residence_country_id).first()
+            nationality = db.session.query(Country).filter(Country.id == user.nationality_country_id).first()
+            date_of_birth = user.user_dateOfBirth
+
+            is_sent = generate(template_path=template_url,
+                               event_id=event_id,
+                               work_address=work_address,
+                               addressed_to=addressed_to,
+                               residential_address=residential_address,
+                               passport_name=passport_name,
+                               passport_no=passport_no,
+                               passport_issued_by=passport_issued_by,
+                               invitation_letter_sent_at=invitation_letter_request.invitation_letter_sent_at,
+                               to_date=to_date,
+                               from_date=from_date,
+                               country_of_residence=country_of_residence,
+                               nationality=nationality,
+                               date_of_birth=date_of_birth,
+                               email=user.email)
+
+            if is_sent:
```

Comment:
if it's not sent, then we also need a failcase. 
"Failed to send Invitation Email"